### DISCUSSION TITLE
ZWEL0029E early at startup

### DISCUSSION THREAD

**Discussion by PavelKondratiev** [zowe/zowe-install-packaging#3925] [Q&A]
Hello team,

I just installed Zowe v2.16 on our zPDT, first start of main task was ok but now it terminates quickly with ZWEL0029E message

```
INFO ZWEL0021I Zowe Launcher starting                                           
INFO -- If you cant see '500' at the end of the line, your log is too short to r
INFO ZWEL0023I Zowe YAML config file is 'FILE(/apps/zowe/v216/instance/zowe.yaml
INFO ZWEL0024I HA_INSTANCE_ID is '__ha_instance_id__'                           
INFO ZWEL0017I ROOT_DIR is '/apps/zowe/v216/instance'                           
INFO ZWEL0069I Configuration is valid                                           
INFO ZWEL0058I WORKSPACE_DIR is '/apps/zowe/v216/workspace'                     
INFO ZWEL0018I Zowe instance prepared successfully                              
ERROR ZWEL0029E start component list is empty
```

I did no changes in configuration inbetween successful and unsuccessful runs
Any suggestions are gratefully appreciated!
Thanks in advance! 

Pavel

**Comment by Martin-Zeithaml** [comment]
Could you verify, if the launcher starts with [`HEAPPOOLS64(OFF)`](https://github.com/zowe/launcher/blob/8a2fc186223af82f7fbdeda9855b659fb9c18871/samplib/ZWESLSTC#L59)?

**Comment by PavelKondratiev** [comment]
Yes, it does

```
 VIEW       ADCD.Z25B.PROCLIB(ZWESLSTC) - 01.00             Columns 00001 00072
 000034 //*                                                                    
 000035 //* Or a list of FILE() or PARMLIB() and colon : separated paths       
 000036 //*   in the form of                                                   
 000037 //*                                                                    
 000038 //*    +------------ : ------------+                                   
 000039 //*    V                           |                                   
 000040 //* >--+--FILE(ussPath)------------+--><                               
 000041 //*    |                           |                                   
 000042 //*    +--PARMLIB(dsname(member))--+                                   
 000043 //*                                                                    
 000044 //* Example:                                                           
 000045 //*   CONFIG=FILE(/my/long/path/to/1.yaml)\                            
 000046 //*   :PARMLIB(ZOWE.PARMLIB(YAML))                                     
 000047 //*                                                                    
 000048 //* In the above case, the \ is used as a line continuation.           
 000049 //*                                                                    
 000050 //* When using a list, files on left override properties               
 000051 //* from files to their right.                                         
 000052 //* Typically the right-most file should be the Zowe default yaml.     
 000053 //*                                                                    
 000054 //* Note: All PARMLIB() entries must all have the same member name.    
 000055 //*                                                                    
 000056 //********************************************************************/
 000057 //STDENV   DD  *                                                       
 000058 _CEE_ENVFILE_CONTINUATION=\                                            
 000059 _CEE_RUNOPTS=HEAPPOOLS(OFF),HEAPPOOLS64(OFF)                           
 000060 _EDC_UMASK_DFLT=0002                                                   
 000061 CONFIG=/apps/zowe/v216/instance/zowe.yaml                              
 000062 /*                                                                     
```

**Comment by PavelKondratiev** [comment]
One more question, please - What are the possible conditions that may cause ZWEL0029E to appear? 
Btw I could not find message description here - https://docs.zowe.org/stable/troubleshoot/launcher/launcher-error-codes/

Thanks in advance!

**Comment by Martin-Zeithaml** [comment]
There should be only one possible condition, no components are enabled. Let's show this on the simplified yaml:
```yaml
zowe:
# setup datasets, certificates etc.
  setup:
    dataset:
# ...
# general setting
  runtimeDirectory: /path/to/zowe
# ...
# And components, which could be enabled or disabled
components:
  gateway:
    enabled: false
  app-server:
     enabled: false
  another-component:
    enabled: false
# And so on
```

An attempt to run zowe with this yaml must end with `ZWEL0029E`, because all components are disabled (`enable = false`).

If you have at least 1 component enabled, then it is a bug.

Related to the doc problem, there is a pull request to document missing codes, the suggestion from the pull request says:
_Review the components defined in configuration._

**Comment by 1000TurquoisePogs** [comment]
If the previous comment isnt the solution, can you add "ZLDEBUG=ON" as a line in the ZWESLSTC JCL, such as above or below the CONFIG line? That will turn some debugging on. You can turn on even more debugging by turning this "info" to "trace" https://github.com/zowe/zowe-install-packaging/blob/1f9ad776820f62672d7b3e17d64ef8b9b0569332/example-zowe.yaml#L411-L413
Anyway, that would help us determine what happens in the events leading up to ZWEL029E.

**Comment by PavelKondratiev** [comment]
Hello support,

Thanks for the suggestion, here are the logs
[Debug.zip](https://github.com/user-attachments/files/16447379/Debug.zip)

**Comment by Martin-Zeithaml** [comment]
Could you try to start the Zowe with `zowe.useConfigmgr = false`? It will increase the startup time (it is highly recommenced to use `zowe.useConfigmgr = true`), but it could possibly help us to eliminate certain code path.

**Comment by PavelKondratiev** [comment]
Hi Martin, 

I am re-installed product from scratch and issue is gone so I am closing this discussion.
Please, take a look on my new questions from https://github.com/zowe/zowe-install-packaging/discussions/3930
Thanks!
